--- a/drivers/mtd/spi-nor/core.c
+++ b/drivers/mtd/spi-nor/core.c
@@ -3224,8 +3224,10 @@ static int spi_nor_init(struct spi_nor *
 		 * reboots (e.g., crashes). Warn the user (or hopefully, system
 		 * designer) that this is bad.
 		 */
+		/*
 		WARN_ONCE(nor->flags & SNOR_F_BROKEN_RESET,
 			  "enabling reset hack; may not recover from unexpected reboots\n");
+		*/
 		err = spi_nor_set_4byte_addr_mode(nor, true);
 		if (err)
 			return err;
