diff -ur sx1302-hal-2.1.0/tools/reset_lgw.sh sx1302-hal-2.1.0-mod/tools/reset_lgw.sh
--- sx1302-hal-2.1.0/tools/reset_lgw.sh	2021-04-22 20:34:46.000000000 +0800
+++ sx1302-hal-2.1.0-mod/tools/reset_lgw.sh	2021-12-24 21:16:35.933261876 +0800
@@ -12,10 +12,10 @@
 # GPIO mapping has to be adapted with HW
 #
 
-SX1302_RESET_PIN=23     # SX1302 reset
-SX1302_POWER_EN_PIN=18  # SX1302 power enable
-SX1261_RESET_PIN=22     # SX1261 reset (LBT / Spectral Scan)
-AD5338R_RESET_PIN=13    # AD5338R reset (full-duplex CN490 reference design)
+SX1302_RESET_PIN=497     # GPIO17 SX1302 reset
+SX1302_POWER_EN_PIN=459  # GPIO43 SX1302 power enable
+SX1261_RESET_PIN=460     # GPIO44 SX1261 reset (LBT / Spectral Scan)
+AD5338R_RESET_PIN=461    # GPIO45 AD5338R reset (full-duplex CN490 reference design)
 
 WAIT_GPIO() {
     sleep 0.1
