diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.AS923.USB sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.AS923.USB
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.AS923.USB	2021-04-22 20:34:46.000000000 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.AS923.USB	2022-01-13 11:01:08.408338115 +0800
@@ -98,7 +98,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.CN490 sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.CN490
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.CN490	2022-01-13 10:42:55.087254790 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.CN490	2022-01-13 11:01:08.408338115 +0800
@@ -77,7 +77,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.CN490.USB sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.CN490.USB
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.CN490.USB	2021-04-22 20:34:46.000000000 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.CN490.USB	2022-01-13 11:01:08.344341868 +0800
@@ -102,7 +102,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.EU868 sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.EU868
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.EU868	2022-01-13 10:42:55.087254790 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.EU868	2022-01-13 11:01:08.344341868 +0800
@@ -102,7 +102,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.EU868.USB sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.EU868.USB
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.EU868.USB	2021-04-22 20:34:46.000000000 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.EU868.USB	2022-01-13 11:01:08.408338115 +0800
@@ -101,7 +101,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.US915 sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.US915
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.US915	2022-01-13 10:42:55.087254790 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.US915	2022-01-13 11:01:08.408338115 +0800
@@ -89,7 +89,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.US915.USB sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.US915.USB
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1250.US915.USB	2021-04-22 20:34:46.000000000 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1250.US915.USB	2022-01-13 11:01:08.340342104 +0800
@@ -88,7 +88,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1255.CN490.full-duplex sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1255.CN490.full-duplex
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1255.CN490.full-duplex	2022-01-13 10:42:55.087254790 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1255.CN490.full-duplex	2022-01-13 11:01:08.408338115 +0800
@@ -65,7 +65,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
diff -ur sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1257.EU868 sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1257.EU868
--- sx1302-hal-2.1.0/packet_forwarder/global_conf.json.sx1257.EU868	2022-01-13 10:42:55.087254790 +0800
+++ sx1302-hal-2.1.0-mod/packet_forwarder/global_conf.json.sx1257.EU868	2022-01-13 11:01:08.344341868 +0800
@@ -71,7 +71,7 @@
         "forward_crc_error": false,
         "forward_crc_disabled": false,
         /* GPS configuration */
-        "gps_tty_path": "/dev/ttyS0",
+        /* "gps_tty_path": "/dev/ttyS0", */
         /* GPS reference coordinates */
         "ref_latitude": 0.0,
         "ref_longitude": 0.0,
