--- linux-5.10.179/arch/mips/ralink/mt7620.c	2023-05-23 18:36:50.569303158 +0800
+++ linux-5.10.179/arch/mips/ralink/mt7620.c.mod	2023-05-23 18:39:57.160063046 +0800
@@ -711,8 +711,16 @@
 	pr_info("Digital PMU set to %s control\n",
 		(pmu1 & DIG_SW_SEL) ? ("sw") : ("hw"));
 
-	if (is_mt76x8())
+	if (is_mt76x8()){
+		// Following 5 line copy from drivers/mmc/host/mtk-mmc/sd.c msdc_drv_probe(struct platform_device *pdev)
+		//FIXME: this should be done by pinconf and not by the sd driver
+		if ((ralink_soc == MT762X_SOC_MT7688 ||
+		ralink_soc == MT762X_SOC_MT7628AN) &&
+		(!(rt_sysc_r32(0x60) & BIT(15))))
+		rt_sysc_m32(0xf << 17, 0xf << 17, 0x3c);
+
 		rt2880_pinmux_data = mt7628an_pinmux_data;
+	}
 	else
 		rt2880_pinmux_data = mt7620a_pinmux_data;
 }
