--- linux-5.15.150/drivers/char/random.c	2024-03-01 20:22:01.000000000 +0800
+++ linux-5.15.150/drivers/char/random.c.mod	2024-03-11 23:41:36.571216503 +0800
@@ -87,8 +87,10 @@
 static RAW_NOTIFIER_HEAD(random_ready_chain);
 
 /* Control how we warn userspace. */
+/*
 static struct ratelimit_state urandom_warning =
 	RATELIMIT_STATE_INIT_FLAGS("urandom_warning", HZ, 3, RATELIMIT_MSG_ON_RELEASE);
+*/
 static int ratelimit_disable __read_mostly =
 	IS_ENABLED(CONFIG_WARN_ALL_UNSEEDED_RANDOM);
 module_param_named(ratelimit_disable, ratelimit_disable, int, 0644);
@@ -723,10 +725,12 @@
 		process_random_ready_list();
 		wake_up_interruptible(&crng_init_wait);
 		kill_fasync(&fasync, SIGIO, POLL_IN);
+/*
 		pr_notice("crng init done\n");
 		if (urandom_warning.missed)
 			pr_notice("%d urandom warning(s) missed due to ratelimiting\n",
 				  urandom_warning.missed);
+*/
 	} else if (orig < POOL_EARLY_BITS && new >= POOL_EARLY_BITS) {
 		spin_lock_irqsave(&base_crng.lock, flags);
 		/* Check if crng_init is CRNG_EMPTY, to avoid race with crng_reseed(). */
@@ -1283,6 +1287,7 @@
 
 static ssize_t urandom_read_iter(struct kiocb *kiocb, struct iov_iter *iter)
 {
+/*
 	static int maxwarn = 10;
 
 	if (!crng_ready()) {
@@ -1294,7 +1299,7 @@
 				  current->comm, iov_iter_count(iter));
 		}
 	}
-
+*/
 	return get_random_bytes_user(iter);
 }
 
